lib/bankaccounts/savings-account.rb

```diff
@@ -0,0 +1,27 @@
+module BankAccounts
+
+  # SavingsAccount inherits from Account class
+  class SavingsAccount < Account
+    def initialize(id, initial_balance)
+      super
+      @min_balance = 10
+      check_initial_balance(initial_balance)
+      @fee = 2
+    end
+
+    # withdraw is inherited from Account; therefore, it does not appear here.
+    # withdraw(amount) subtracts amount from balance, incurs $2 fee.
+    # if account would fall below @min_balance, outputs warning & returns unmodified balance.
+
+    # Calculates interest and adds it to the balance.
+    def add_interest(rate)
+      # Calculates interest & updates balance
+      interest = @balance * (rate/100)
+      @balance += interest
+      # Returns calculated interest _not_ the updated balance
```

Comment:
Great comment. This is inconsistent with the other methods (everything else returns the modified balance), so having a comment here clarifies things a lot.